rfcs/react-components/non-core-components.md

```diff
@@ -0,0 +1,388 @@
+
+The extent of 'deviating' will have to depend on the requirements of each component, but we should be ready to accept
+that we might end up delivering a set of components that don't follow the principles that drive the core library
+(@fluentui/react-components).
+
+In cases where components need to deviate, we should notify the team through specs and architecture meetings
+(i.e. tech sync), what deviations are done and the context behind them. All non-core components should still
+follow the V9 spec driven approach.
+
+#### Pros
+
+- 👍 Can enforce design across Microsoft more strictly
+- 👍 Can be more flexible than core components
+- 👍 Can be done on a case-by-case basis depending on requirements
+
+#### Cons
+
+- 👎 Can lead to a very inconsistent set of components that behave differently to each other
+- 👎 Our partners might still want to customize to the max
+- 👎 Approaches could be heavily biased on a subset of partners
+
+### Breaking change cadence
+
+#### Same cadence as core components
+
+Breaking changes for non-core components will follow the same cadence as core components i.e.
+breaking changes can only happen when @fluentui/react-components reaches v10.
+
+##### Pros
+
+- 👍 Predictability - nothing new compared to current library
+- 👍 Good support for partners to build on
+
+##### Cons
+
+- 👎 Partners might not mind breaking changes for specialized components that are not used often
+- 👎 Maintaining legacy for components that are not used often in applications
+
+#### Faster than core components
+
+Breaking changes for non-core components can occur more frequently than for core components. The proposal does not
+mean that breaking changes will be taken lightly, or that there will be a requirement to make breaking changes often.
+
+##### Pros
+
+- 👍 Bleeding edge improvements for partners
+- 👍 Specialized components might not have widespread use and can be upgraded in a few instances
+
+##### Cons
+
+- 👎 Cost of breaking changes has not changed
+- 👎 Just because we make breaking changes does not mean users will upgrade earlier
+- 👎 Need to handle urgent bugfixes for most past versions - need a clear support SLA
+
+### Choosing a repo for
+
+#### Current monorepo
+
+Non-core packages should still live in the monorepo.
+
+##### Pros
+
+- 👍 Easier setup
+- 👍 Nothing new to learn
+- 👍 Issue reporting path does not change
+
+##### Cons
+
+- 👎 We might not want the same release cadence - releasing a fix for a non-core component would release all of core
+- 👎 Extra load on the CI as more non-core components are needed
+- 👎 Not possible to have more separation from core components without lots of tooling hacks
+
+#### Separate monorepo
+
+We can create a new monorepo for our non-core packages and depend on core packages from NPM.
+
+##### Pros
+
+- 👍 We can have our ideal DX
+- 👍 Release cadence does not need to depend on core
+- 👍 Developing our with our public API - we eat our own dogfood
```

Comment:
this is better than a single monorepo, but is this repo public? private? do we need 1 of each?